### PR TITLE
ACR: keep uploads with BinaryData only

### PR DESCRIPTION
**Pull request by lmolkova**
~~We can offer an option to make content replayable in future, for now defaulting to replayable content only.~~

### CODE REVIEW

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryContentAsyncClient.java [hunk 1]
```diff
@@ -209,7 +206,7 @@ public Mono<Response<SetManifestResult>> setManifestWithResponse(SetManifestOpti
      * </pre>
      * <!-- end com.azure.containers.containerregistry.uploadBlobAsyncErrorHandling -->
      *
-     * Content is uploaded in chunks of up to 4MB size. Chunk size depends on passed {@link ByteBuffer}
+      * Content is uploaded in chunks of up to 4MB size. Chunk size depends on passed {@link ByteBuffer}
```
Comment (alzimmermsft) [inline sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryContentAsyncClient.java:209]: This should be based on the passed BinaryData, not ByteBuffer right?

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryContentClient.java [hunk 2]
```diff
@@ -236,9 +223,21 @@ public UploadRegistryBlobResult uploadBlob(BinaryData content) {
      * @throws NullPointerException thrown if the {@code stream} is {@code null}.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public UploadRegistryBlobResult uploadBlob(ReadableByteChannel content, Context context) {
+    public UploadRegistryBlobResult uploadBlob(BinaryData content, Context context) {
         Objects.requireNonNull(content, "'content' cannot be null.");
-        return runWithTracing(UPLOAD_BLOB_SPAN_NAME, (span) -> uploadBlobInternal(content, span), context);
+
+        InputStream stream = content.toStream();
+        ReadableByteChannel channel = Channels.newChannel(stream);
```
Comment (alzimmermsft) [inline sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryContentClient.java:230]: Is converting the BinaryData to a `ReadableByteChannel` the right way to go about this? For now, I think it's fine as this is implementation logic and can be changed in the future.

Comment (lmolkova) [reply sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryContentClient.java]: good point, switched to stream

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**